static/compiler.js

```diff
@@ -601,6 +601,7 @@ Compiler.prototype.onCompileResponse = function (request, result, cached) {
 
     this.compileTimeLabel.text(timeLabelText);
 
+    result.stderr = result.stderr.slice(0, options.truncateErrorMessages);
```

Comment:
Could this be done serverside?

Reply:
Thanks all; I agree -- if possible we should truncate on the server, rather than send multi-megabyte documents only to shorten them on the client. We already have limits on the output size (in bytes); adding a number of lines truncation would be a nice extension.

This is not a deal-breaker though: we can limit in both places, if that makes sense.